src/thunderbots/software/backend_input/networking/ssl_vision_client.cpp

```diff
@@ -0,0 +1,51 @@
+#include "backend_input/networking/ssl_vision_client.h"
+
+SSLVisionClient::SSLVisionClient(const std::string ip_address, const unsigned short port)
+    : socket_(io_service)
+{
+    boost::asio::ip::udp::endpoint listen_endpoint(
+        boost::asio::ip::address::from_string(ip_address), port);
+    socket_.open(listen_endpoint.protocol());
+    socket_.bind(listen_endpoint);
+
+    // Join the multicast group.
+    socket_.set_option(boost::asio::ip::multicast::join_group(
+        boost::asio::ip::address::from_string(ip_address)));
+
+    // Start listening for data
+    socket_.async_receive_from(boost::asio::buffer(raw_received_data_, max_buffer_length),
+                               sender_endpoint_,
+                               boost::bind(&SSLVisionClient::handleDataReception, this,
+                                           boost::asio::placeholders::error,
+                                           boost::asio::placeholders::bytes_transferred));
+}
+
+void SSLVisionClient::handleDataReception(const boost::system::error_code &error,
+                                          size_t num_bytes_received)
+{
+    if (!error)
+    {
+        // We create a new pointer here rather than reset or clear the existing pointer
+        // because the unique_ptr is returned with std::move in the getVisionPacket()
+        // function, meaning we can no longer access that memory. Hence we create a new
+        // pointer to work with
+        packet_data = std::make_unique<SSL_WrapperPacket>();
+        packet_data->ParseFromArray(raw_received_data_,
+                                    static_cast<int>(num_bytes_received));
+
+        // Once we've handled the data, start listening again
+        socket_.async_receive_from(
+            boost::asio::buffer(raw_received_data_, max_buffer_length), sender_endpoint_,
+            boost::bind(&SSLVisionClient::handleDataReception, this,
+                        boost::asio::placeholders::error,
+                        boost::asio::placeholders::bytes_transferred));
+    }
```

Comment:
Do we just do nothing here? Might want to make a TODO to throw an exception or something

Reply:
Yea it currently does nothing. I was under the assumption if there was an error it would just drop it and move on, but that might not be the case since `async_receive` wouldn't be called again. I'll add a clause to start receiving again and a `TODO` to log a warning

Reply:
done